source/objects/smoke.lua

```diff
@@ -66,7 +66,7 @@ end
 
 function Smoke.draw()
 	-- draw smoke trail
-	for key, particle in ipairs(Smoke.particles) do
+	for key, _ in ipairs(Smoke.particles) do
         local particle = Smoke.particles[key]
```

Comment:
Nope, thats my mistake. Just remove `local particle = Smoke.particles[key]` instead of changing particle to _. this line was used because I previously didn't use ipairs to iterate but instead a basic for loop like 
```lua
for key=1, #Smoke.particles do
local particle = Smoke.particles[key]
end
```
Which can be done, but wasn't needed in that case and I seemingly forgot to remove the local definition.

Reply:
Fixed.